### ISSUE TITLE
Large runs in BULK Scrips is skipping cases

### ISSUE THREAD

**Issue by C-Love** [MN-Script-Team/DHS-MAXIS-Scripts#2970]
A worker reported that when they run BULK REPT/ACTV on a large number of caseloads then run it on those same caseloads individually, there are some cases that are not on the Excel Spreadsheet from the large run. 

It seems the issue is stemming from the resolution for a "ghost" issue that is in a number of scripts. Essentially, every case number is compared to a list of case numbers already found. It is supposed to work that if there is a duplicate, the script skips the rest of the cases on that REPT/ACTV page. The problem comes in if a case number is enclosed in another case number. Then the fix registers a 'ghost' when there isn't one. 

Here are the scripts that I have found that uses this code and could potentially be creating incomplete reports:

**BULK ACTIONS**
Case Transfer
Targeted SNAP Review Selection

**BULK LISTS**
EOMC
GRMR
INAC
MAMS
MFCM
MONT
MRSR
PND1
PND2
REVS
REVW

**ENCHANCED LISTS**
EXP SNAP Review
Housing Grant Exemption Finder
Spenddown Report

I have a fix ready for this.

**Comment by C-Love** [comment]
So... apparently REPT/GRMR no longer exists in MAXIS? Anyone know anything about this?

**Comment by Farleight** [comment]
Changed in 07/16:

On Friday, July 1 the GRH Six Month reporting task will be installed into production. Effective in the footer month of 07/01/16 workers will see the following changes:

1)    GRH PIC has been added to the STAT JOBS panel.

2)    STAT REVW Cash/GRH will include a selection field for a Pop-Up for entry of GRH six month reporting and ER dates.

3)    STAT MONT reference to GRH will be removed.

4)    REPT GRMR will no longer be valid.

5)    WB has been removed from STAT REVW and REPT REVW.

6)    HRF’s will no longer be mailed for GRH clients with earned income. 

7)    Non-SSI and Non- MFIP ELIG types with earned income of $100 or more will have a SIX MONTH reporting status and mandatory pre-pay code. 

 

Six month reporting for GRH will affect the clients that were monthly reporters. GRH will no longer support monthly reporting beginning in the July footer month.  Months prior to July will continue to require monthly reporting for pending applications and reinstatements with earned income of $100.00 or more.  

If you are processing a HRF for a monthly reporter for June, this is the last HRF they will receive. Update STAT MONT in 06/16 and enter income and other reported information for 06/16. Once eligibility for June is determined, pay can be released. 

The next action for cases with $100.00 or more of ongoing earned income, is to update the STAT JOBS by entering the prospective income for 07/16 and 08/16. In addition, workers will be required to enter income in the GRH PIC, the new pop-up found at the bottom right hand corner of the STAT JOBS panel.  Refer to CM 0007.03.02 for policy information on how income should be entered. 

On STAT REVW, enter a six month review date in the new pop-up window for GRH Six month reporters. To determine the SR date to enter, count six months prior to the ER date. If the date is 09/16 or later enter that date in the SR field. If the date is earlier than 09/16 enter the date that is six months after the ER date. If the GRH applicant or recipient is receiving SNAP and is six month reporting, use the same SR date as SNAP. A “Review Cheat Sheet” is included on page 6 of the GRH 6 Month Budgeting and Reporting PP Handout, see link below. 

Once the PIC is updated and STAT REVW has an SR date, the GRH result should reflect six month reporting on the GRSM panel. The budget will reflect the total of earned income from BUSI, RBIC and all PIC totals from STAT JOBS as well as any unearned income. Review the budget for accuracy. The budget will be set as pre-paid for July and August. July payment will be issued the night of the approval and August will be sent out in the monthly issuance job. 

This payment method will appear as Pre-Pay for Six Month reporters. Workers should not FIAT to post pay unless the client is projected to be in the GRH setting for less than 30 days or the worker has information that the client is leaving the facility in the following month or is no longer GRH eligible. 

For more information please refer to CM0007.03.02 and Bulletin 16-48-03 GRH 6 Month Reporting.  POLI TEMP updates effective 07/16 include, TE18.11 GRH-OVERVIEW PART 2, TE18.19 BUDGETING FOR GRH SIX-MONTH REPORTING, TE18.18 GRH SIX MONTH REVIEW DATES, and TE02.07.366 REPORT FORMS MAILING

**Comment by C-Love** [comment]
Right... I remember that now. You are a veritable font of knowledge. Thank you. I will create a new issue to remove that script.

**Comment by C-Love** [comment]
I have a question about this fix. I have fixed these scripts already - however in the course of fixing them, I have also added some enhancements. 

Enhancements added:
-Functionality so that if a supervisor X-Number is entered in the caseloads to check, all of the x-numbers listed under that supervisor in MAXIS will be added to the list to check
-For reports that check PND2, the script will identify and note when a case has multiple applications pending
-Handling for reports in which one case has multiple lines due to multiple people

Would it be alright to include these enhancements (I would create separate issues for them so they are documented)? It would make the changes faster and easier as I can just copy the files that are already updated and tested.

**Comment by C-Love** [comment]
I have tested these, the fix is working.
